### PR TITLE
Bump futures and tokio to stable releases

### PR DESCRIPTION
**Pull request by jgrund**
Move tokio and futures to stable releases.

This is a large PR as there were many small API changes required across
the workspace.

Signed-off-by: Joe Grund <jgrund@whamcloud.io>

### CODE REVIEW

**Comment by utopiabound** [comment]
Tokio 0.2.5 was released late yesterday.

**Comment by jgrund** [comment]
> Tokio 0.2.5 was released late yesterday.

Yes, I'm purposely pinning to 0.2.4, as the latest warp doesn't support 0.2.5 yet.

File: iml-agent/Cargo.toml [hunk 1]
```diff
@@ -8,28 +8,24 @@ license = "MIT"
...
-futures-preview = "0.3.0-alpha.19"
+futures = "0.3"
 lazy_static = "1.4.0"
 termion = "1"
 spinners = "1.0.0"
 libc = "0.2"
 prettytable-rs = "0.8"
-reqwest = { version = "0.10.0-alpha.2", features = ["default-tls", "tls", "json", "unstable-stream"] }
-http = "0.1"
+reqwest = { git = "https://github.com/seanmonstar/reqwest", features = ["default-tls", "tls", "json", "stream"] }
```
[the remark this answers is not in the document]
Comment (jgrund) [reply iml-agent/Cargo.toml]: I don't want to pin in the `Cargo.toml` for now. I think using a git ref should be fairly short-lived.

**Comment by ip1981** [comment]
Now I get some errors:

```
error[E0277]: the trait bound `tokio::fs::read_dir::ReadDir: futures_core::stream::Stream` is not satisfied
   --> iml-fs/src/lib.rs:124:10
    |
124 |         .try_flatten_stream()
    |          ^^^^^^^^^^^^^^^^^^ the trait `futures_core::stream::Stream` is not implemented for `tokio::fs::read_dir::ReadDir`
```

**Pull request merged**